### ISSUE TITLE
FluxStore 1.5.6 Category Icon Issue

### ISSUE THREAD

**Issue by DrChrispoper** [inspireui/support#1341]
> - Thank you for reaching please follow the format below to create a new ticket, please explain as more detail as possible and allow 24 - 48 hours to receive an answer to your ticket.
> - Try to explain your all problems in one ticket, instead of creating multi tickets 
> - After creating an issue, please copy the URL to the Product link on http://codecanyon.net - to help us know you have purchased our product (example: https://tppr.me/DJA8L)
> - IMPORTANT: We will close the ticket if you could not prove as our customer.

---------

1. My issue
In 1.5.6 you have updated the widget category_icon_items.dart from
```onTap: () => Product.showList(config: item, context: context),```

to
```
onTap: () => Product.showList(
                  config: item,
                  context: context,
                  products: item['data']  ?? [] ,
                ),
```
Now the product list doesn't show as item has not 'data'.

2. My issue
In the magento.dart

You added
`kAdvanceConfig["EnableConfigurableProduct"]`

But you did not add it to the config.dart file.

3. My issue
In magento.dart if a product with no category is fetched, the app will crash when you parse it from the json. You need to change:

`product.categoryId = int.parse(categoryIds[0]);`

to

`product.categoryId = categoryIds.length > 0 ? int.parse(categoryIds[0]) : 0;
`

4. My issue
My sugguestion to properly update the magento cart when adding products to the cart.
Change this to properly delete the cart items.
```
Future<bool> addToCart(CartModel cartModel, String token, quoteId,
      {isDelete = false}) async {
    try {
      //delete items in cart
      if (isDelete) {
        await Future.forEach(cartModel.productsInCart.keys, (key) async {
          var productId;
          if (key.contains("-")) {
            productId = int.parse(key.split("-")[0]);
          } else {
            productId = int.parse(key);
          }
          await http.delete(
              "$domain/index.php/rest/V1/carts/mine/items/$productId",
              headers: {'Authorization': 'Bearer ' + token});
        });
      }
      //add items to cart
      await Future.forEach(cartModel.productsInCart.keys, (key) async {
...
        params["quote_id"] = quoteId;
...
  }
```

to

```
  Future<bool> addToCart(CartModel cartModel, String token, cartInfo,
      {isDelete = false}) async {
    try {
      //delete items in cart
      if (isDelete) {
        await Future.forEach(cartInfo["items"], (item) async {
          await http.delete(
              "$domain/index.php/rest/V1/carts/mine/items/${item["item_id"]}",
              headers: {'Authorization': 'Bearer ' + token});
        });
      }
      //add items to cart
      await Future.forEach(cartModel.productsInCart.keys, (key) async {
...
        params["quote_id"] = cartInfo["id"];
...
```

And update this following function to match it

`  Future<bool> addItemsToCart(CartModel cartModel, String token) async {
`

From

```
...
        return await addToCart(cartModel, token, cartInfo["id"],
...
          final quoteId = convert.jsonDecode(res.body);
          return await addToCart(cartModel, token, quoteId);
...
```

to

```
...
        return await addToCart(cartModel, token, cartInfo,
...
          final cartInfo = convert.jsonDecode(res.body);
          return await addToCart(cartModel, token, cartInfo);
...
```

**Comment by minhtruong315** [comment]
@DrChrispoper send me config.json, config.dart file and admin account via email: truong@inspireui.com
with subject: https://github.com/inspireui/support/issues/1341

**Comment by DrChrispoper** [comment]
Hi,
I've sent you the files.
What admin account are you referring to?
Thanks.

**Comment by DrChrispoper** [comment]
5. My Issue
In config.dart (I am using Magento)
When I set "EnableConfigurableProduct" to true, the system only fetches configurable products.
When I set it to false; I still set fetch the configurable products and all the others. 

The other issue is the prices displayed for the configurable products, in either case, is 0 in the list view but the price appears correctly on the detail page.

**Comment by minhtruong315** [comment]
Fluxstore hasn't supported load configurable product and single product together if EnableConfigurableProduct is true. We will update it in the future.Thanks

**Comment by DrChrispoper** [comment]
Ok Thanks.
Anything on the other issue?

**Comment by minhtruong315** [comment]
which issues? can you tell me the detail?

**Comment by DrChrispoper** [comment]
> > * Thank you for reaching please follow the format below to create a new ticket, please explain as more detail as possible and allow 24 - 48 hours to receive an answer to your ticket.
> > * Try to explain your all problems in one ticket, instead of creating multi tickets
> > * After creating an issue, please copy the URL to the Product link on http://codecanyon.net - to help us know you have purchased our product (example: https://tppr.me/DJA8L)
> > * IMPORTANT: We will close the ticket if you could not prove as our customer.
> 
> 1. My issue
>    In 1.5.6 you have updated the widget category_icon_items.dart from
>    `onTap: () => Product.showList(config: item, context: context),`
> 
> to
> 
> ```
> onTap: () => Product.showList(
>                   config: item,
>                   context: context,
>                   products: item['data']  ?? [] ,
>                 ),
> ```
> 
> Now the product list doesn't show as item has not 'data'.
> 
> 1. My issue
>    In the magento.dart
> 
> You added
> `kAdvanceConfig["EnableConfigurableProduct"]`
> 
> But you did not add it to the config.dart file.
> 
> 1. My issue
>    In magento.dart if a product with no category is fetched, the app will crash when you parse it from the json. You need to change:
> 
> `product.categoryId = int.parse(categoryIds[0]);`
> 
> to
> 
> `product.categoryId = categoryIds.length > 0 ? int.parse(categoryIds[0]) : 0; `
> 
> 1. My issue
>    My sugguestion to properly update the magento cart when adding products to the cart.
>    Change this to properly delete the cart items.
> 
> ```
> Future<bool> addToCart(CartModel cartModel, String token, quoteId,
>       {isDelete = false}) async {
>     try {
>       //delete items in cart
>       if (isDelete) {
>         await Future.forEach(cartModel.productsInCart.keys, (key) async {
>           var productId;
>           if (key.contains("-")) {
>             productId = int.parse(key.split("-")[0]);
>           } else {
>             productId = int.parse(key);
>           }
>           await http.delete(
>               "$domain/index.php/rest/V1/carts/mine/items/$productId",
>               headers: {'Authorization': 'Bearer ' + token});
>         });
>       }
>       //add items to cart
>       await Future.forEach(cartModel.productsInCart.keys, (key) async {
> ...
>         params["quote_id"] = quoteId;
> ...
>   }
> ```
> 
> to
> 
> ```
>   Future<bool> addToCart(CartModel cartModel, String token, cartInfo,
>       {isDelete = false}) async {
>     try {
>       //delete items in cart
>       if (isDelete) {
>         await Future.forEach(cartInfo["items"], (item) async {
>           await http.delete(
>               "$domain/index.php/rest/V1/carts/mine/items/${item["item_id"]}",
>               headers: {'Authorization': 'Bearer ' + token});
>         });
>       }
>       //add items to cart
>       await Future.forEach(cartModel.productsInCart.keys, (key) async {
> ...
>         params["quote_id"] = cartInfo["id"];
> ...
> ```
> 
> And update this following function to match it
> 
> ` Future<bool> addItemsToCart(CartModel cartModel, String token) async {`
> 
> From
> 
> ```
> ...
>         return await addToCart(cartModel, token, cartInfo["id"],
> ...
>           final quoteId = convert.jsonDecode(res.body);
>           return await addToCart(cartModel, token, quoteId);
> ...
> ```
> 
> to
> 
> ```
> ...
>         return await addToCart(cartModel, token, cartInfo,
> ...
>           final cartInfo = convert.jsonDecode(res.body);
>           return await addToCart(cartModel, token, cartInfo);
> ...
> ```

These 4.

**Comment by DrChrispoper** [comment]
Where are you checking the issues from?

**Comment by minhtruong315** [comment]
1. I'm working on latest version. Please update then check again, if it still have issues, please record screen.
2. I see it is added on config.dart. please check again
<img width="833" alt="Screen Shot 2019-11-16 at 9 27 50 AM" src="https://user-images.githubusercontent.com/10402500/68986619-a282a480-0853-11ea-9f62-97d3b48449a4.png">

**Comment by DrChrispoper** [comment]
Thanks for the update.
1. 


2. I've just checked your latest update on Gitlab and it is still missing for some reason.
![screenshot-gitlab com-2019 11 16-10_04_19](https://user-images.githubusercontent.com/1864825/68991636-91886200-0858-11ea-8083-d2792b17414b.png)

3. Thanks!

4. The issue is that I tested your code and it isn't deleting all items from the cart.
This is for a connected user.
If I have added items to the cart on the website, it will not be displayed (yet) on the app.
So when I am adding items in the cart on the app I want to delete all the items that were added on the website.
If we don't they will be charged for them on checkout even though they cannot see them on the app.

<img width="1440" alt="Screenshot 2019-11-16 at 10 23 14" src="https://user-images.githubusercontent.com/1864825/68997332-1dbc7880-089d-11ea-87c6-75cd6c921765.png">

"cartInfo" contains the information of previously added items to delete but "cartModel" doesn't.
